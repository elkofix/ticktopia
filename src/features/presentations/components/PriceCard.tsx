export function PriceCard({ price }: { price: number }) {
  return (
    <div className="bg-white p-6 rounded-xl shadow-md">
      <div className="flex justify-between items-center">
        <span className="text-lg font-semibold text-gray-700">Asiento General</span>
        <span className="text-xl font-bold text-blue-600">${price.toFixed(2)}</span>
      </div>
    </div>
  );
}