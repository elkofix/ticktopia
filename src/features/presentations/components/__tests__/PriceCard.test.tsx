import React from 'react';
import { render, screen } from '@testing-library/react';
import { PriceCard } from '../PriceCard';
import '@testing-library/jest-dom';

describe('PriceCard', () => {
  it('renders correctly with the provided price', () => {
    const testPrice = 49.99;
    render(<PriceCard price={testPrice} />);

    // Check container
    const container = screen.getByTestId('price-card');
    expect(container).toBeInTheDocument();
    expect(container).toHaveClass('bg-white');
    expect(container).toHaveClass('p-6');
    expect(container).toHaveClass('rounded-xl');
    expect(container).toHaveClass('shadow-md');

    // Check title
    const title = screen.getByText('Asiento General');
    expect(title).toBeInTheDocument();
    expect(title).toHaveClass('text-lg');
    expect(title).toHaveClass('font-semibold');
    expect(title).toHaveClass('text-gray-700');

    // Check price
    const priceElement = screen.getByText(`$${testPrice.toFixed(2)}`);
    expect(priceElement).toBeInTheDocument();
    expect(priceElement).toHaveClass('text-xl');
    expect(priceElement).toHaveClass('font-bold');
    expect(priceElement).toHaveClass('text-blue-600');
  });

  it('formats the price to 2 decimal places', () => {
    const testPrice = 50;
    render(<PriceCard price={testPrice} />);
    
    expect(screen.getByText('$50.00')).toBeInTheDocument();
  });

  it('handles zero price correctly', () => {
    render(<PriceCard price={0} />);
    
    expect(screen.getByText('$0.00')).toBeInTheDocument();
  });

  it('handles large prices correctly', () => {
    const largePrice = 1234567.89;
    render(<PriceCard price={largePrice} />);
    
    expect(screen.getByText('$1234567.89')).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<PriceCard price={49.99} />);
    expect(asFragment()).toMatchSnapshot();
  });
});