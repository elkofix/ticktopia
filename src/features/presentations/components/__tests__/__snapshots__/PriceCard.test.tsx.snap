// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PriceCard matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="bg-white p-6 rounded-xl shadow-md"
    data-testid="price-card"
  >
    <div
      class="flex justify-between items-center"
    >
      <span
        class="text-lg font-semibold text-gray-700"
      >
        Asiento General
      </span>
      <span
        class="text-xl font-bold text-blue-600"
        data-testid="presentation-price"
      >
        $49.99
      </span>
    </div>
  </div>
</DocumentFragment>
`;
